arbor/util/piecewise.hpp

```diff
@@ -484,11 +484,13 @@ inline pw_elements<void> zip(const pw_elements<void>& a, const pw_elements<void>
     double a_right = a.interval(ai).second;
     double b_right = b.interval(bi).second;
 
-    while (left<rmin) {
+    for(;;) {
         double right = std::min(a_right, b_right);
         right = std::min(right, rmin);
 
         z.push_back(left, right);
+        if (right==rmin) break;
+
```

Comment:
I suspect there might be some further dodginess in the zip routines, when it comes to the case of zero-length elements at the end of the common support. What was the error that sanitizing was reporting?

Reply:
If I remember correctly, one level further down, the `second` of a tuple of iterators is dereferenced.